### PR TITLE
🎉 New Source: Kustomer

### LINKED ISSUE #4004: Kustomer connector
## Tell us about the new connector you’d like to have
I'd like to have a new connector to pull data from Kustomer to Snowflake

## Describe the context around this new connector
This is needed to do customer reporting for our marketing team.
The most important endpoints for us are:

- Customers
- Companies
- Customer events
- Conversations
- Notes

Less important but nice to haves would be:
- Teams, Tags, Shortcuts, Users

We would run this sync at least on a daily basis, maybe more frequently.

## Describe the alternative you are considering or using
https://github.com/singer-io/tap-kustomer

### PR DESCRIPTION
**Pull request by rparrapy**
## What
Adds a Kustomer connector.

## How
Wrapping the existing [Singer tap](https://github.com/singer-io/tap-kustomer).

## Pre-merge Checklist
- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [x] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [x] Code reviews completed
- [x] Documentation updated 
    - [ ] `README.md`
    - [x] `docs/SUMMARY.md` if it's a new connector
    - [x] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md` contains a reference to the new connector
    - [x] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [x] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified

This closes #4004

### CODE REVIEW

**Comment by rparrapy** [comment]
I cannot get the acceptance tests for incremental syncs  in _airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py_ to pass, mainly because Singer's state does not conform with what those tests expect.

This is what the connector state looks like:

```
  {
    "bookmarks": {
      "conversations": "2019-12-26T18:35:41.583Z",
      "customers": "2019-12-26T18:39:38.186Z",
      "kobjects": "2019-12-26T18:24:18.283Z",
      "messages": "2019-12-26T18:39:45.018Z",
      "shortcuts": "2019-12-25T18:00:00Z",
      "tags": "2019-12-25T18:00:00Z",
      "users": "2019-12-25T18:00:00Z"
    },
    "currently_syncing": "teams"
  }
```

I noticed that the Github source that is referenced as an example for singer-based connectors does not run acceptance tests either. Is the acceptance test criteria relaxed for Singer-based connectors or else please let me know how to proceed to get this merged.

File: airbyte-integrations/connectors/source-kustomer-singer/source_kustomer_singer/source.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        """
+        url = "https://api.kustomerapp.com/v1/customers"
+        headers = {"Authorization": f"Bearer {config['api_token']}"}
+
+        try:
+            session = requests.get(url, headers=headers)
+            session.raise_for_status()
+            return AirbyteConnectionStatus(status=Status.SUCCEEDED)
+        except requests.exceptions.RequestException as e:
+            return AirbyteConnectionStatus(status=Status.FAILED, message=f"An exception occurred: {str(e)}")
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-kustomer-singer/source_kustomer_singer/source.py:56]:
```suggestion
            return AirbyteConnectionStatus(status=Status.FAILED, message=f"An exception occurred: {repr(e)}")
```

**Review by sherifnada — COMMENTED** [review]
LGTM @rparrapy -- for incremental once https://github.com/airbytehq/airbyte/pull/4552/files is merged you should be able to move forward on this issue (you might need to docker pull the acceptance test image) 

then can you post the output for the remaining tests?

**Comment by rparrapy** [comment]
Finally got some time to wrap this up.

CleanShot 2021-07-21 at 23 05 32@2x

**Comment by rparrapy** [comment]
If we could get this merged this week based on the green integration tests that'd be great for my current project.

**Comment by sherifnada** [comment]
Apologies @rparrapy - have been slammed. Will do it tonight!

**Comment by sherifnada** [comment]
@rparrapy this has been published. Thank you for the contribution!! 🎉

**Pull request merged**